utils/agreement-utils/src/agreement.rs

```diff
@@ -143,6 +144,16 @@ impl TryFrom<&Agreement> for AgreementView {
     }
 }
 
+impl<'a> std::fmt::Display for AgreementView {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FormatError> {
+        // Display not pretty version as fallback.
+        match serde_json::to_string_pretty(&flatten_value(self.json.clone())) {
```

Comment:
no, we do not want it that flat:

```
  "demand.properties.golem.com.payment.chosen-platform": "zksync-rinkeby-tglm",
  "demand.properties.golem.com.payment.debit-notes.accept-timeout?": 120,
  "demand.properties.golem.com.payment.platform.zksync-rinkeby-tglm.address": "0xc44ac5e25c9f91be72d0412e13c6aaa968f63790",
  "demand.properties.golem.node.debug.subnet": "zima",
  "demand.properties.golem.srv.caps.multi-activity": true,
  "demand.properties.golem.srv.comp.expiration": 1612911405147,
  "demand.properties.golem.srv.comp.task_package": "hash:sha3:9a3b5d67b0b27746283cb5f287c13eab1beaa12d92a9f536b747c7ae:http://3.249.139.167:8000/local-image-c76719083b.gvmi\n",
  "demand.properties.golem.srv.comp.vm.package_format": "gvmkit-squash",
  "demand.requestorId": "0xc44ac5e25c9f91be72d0412e13c6aaa968f63790",
  "demand.timestamp": "2021-02-09T22:42:24.079022Z",
  "offer.constraints": "(&\n  (golem.srv.comp.expiration>1612910542713)\n  (golem.node.debug.subnet=zima)\n)",
  "offer.offerId": "a08a279a78df4d62803c9cff1d8f0d31-5f8788089feeaa3a46d045e4fc332e9e59aa09af92262c406926710278cace73",
  "offer.properties.golem.activity.caps.transfer.protocol": [
    "gftp",
    "http",
    "https"
  ],
  "offer.properties.golem.com.payment.debit-notes.accept-timeout?": 120,
  "offer.properties.golem.com.payment.platform.erc20-rinkeby-tglm.address": "0x1231231231231231231231231231231231231231",
  "offer.properties.golem.com.payment.platform.zksync-rinkeby-tglm.address": "0x1231231231231231231231231231231231231231",
  "offer.properties.golem.com.pricing.model": "linear",
  "offer.properties.golem.com.pricing.model.linear.coeffs": [
    5.555555555555556e-6,
    0.00002777777777777778,
    0.0
  ],
  "offer.properties.golem.com.scheme": "payu",
  "offer.properties.golem.com.scheme.payu.interval_sec": 6.0,
  "offer.properties.golem.com.usage.vector": [
  ```
  
  Just flatten properties, and not the whole agreement